### DISCUSSION TITLE
Type of Decorators

### DISCUSSION THREAD

**Discussion by paullewisn** [storybookjs/storybook#25959] [Help] [answered]
### Summary

I have moved some of my story specific Decorators to individual components (at `./storybook/decorators`)

```
export const Decorator = (Story: any) => {
  return (
    <div className={styles.wrapper}>
      <Story />
    </div>
  );
};
```

I want to learn the type of this export, it is listed as:

```
DecoratorFunction<ReactRenderer, T> | DecoratorFunction<ReactRenderer, T>[] | undefined
```

But `DecoratorFunction` does not appear to be exposed. Is there an appropriate type?

**Comment by Marklb** [comment, marked as answer]
It is exported from '@storybook/types', so `import { DecoratorFunction } from '@storybook/types';` should work.

The type is actually defined in '@storybook/csf', but I would use the re-export from '@storybook/types', to make sure you are referring to the same one that your Storybook version is using.
https://github.com/ComponentDriven/csf/blob/cdc80df4da92aa5d28e3d057ff16077e461f2791/src/story.ts#L180-L183

**Reply by paullewisn** [reply]
Thanks 🙏🏼 I have gone with

```
import React from 'react';
import styles from './DarkBackground.module.scss';
import { DecoratorFunction } from '@storybook/types';

export const Decorator: DecoratorFunction<any> = (Story) => {
  return (
    <div className={styles.wrapper}>
      <Story />
    </div>
  );
};
```

as without the generic `<any>` it still errors 👍🏼
